nasa/harmony-regression-tests
# DAS-1588 - Add HOSS regression test for collection with bounds variables

## Description

This PR adds a regression test for HOSS to test changes for DAS-1588. The test makes a spatial (bounding box) and temporal subset request for a collection with bounds variables associated with the dimension variables.

The test uses `xarray` to compare output to a reference image, as first introduced to this notebook by @flamingbear.

~~This test will likely pass using a version of HOSS that either has the bounds branch of code, or prior to that. The GPM/IMERG data have well behaved data that have dimension values at the pixel centres, so HOSS behaves the same way with either version.~~ This is not true due to a bug that will be fixed by the current HOSS DAS-1588 PR.

## Jira Issue ID

DAS-1588

## Local Test Steps

* Pull this branch locally.
* Pull the DAS-1588-bounds branch of the HOSS repository (or master if the PR has been merged)
* Build the HOSS service image from the HOSS repository.
* Restart the Harmony services running in you local Harmony instance.
* Point the `harmony_host_url` to `http://localhost:3000` near the beginning of the notebook.
* Execute the notebook, paying particular attention to all Variable Subsetter and HOSS regression tests. These tests should all pass.

## PR Acceptance Checklist
